bindings/rust/bench/src/s2n_tls.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    harness::{read_to_bytes, Mode, TlsBenchHarness},
+    CA_CERT_PATH, SERVER_CERT_CHAIN_PATH, SERVER_KEY_PATH,
+};
+use log::debug;
+use s2n_tls::{
+    callbacks::VerifyHostNameCallback,
+    config::{Builder, Config},
+    connection::Connection,
+    enums::Blinding,
+    security::DEFAULT_TLS13,
+};
+use std::{
+    cell::UnsafeCell,
+    collections::VecDeque,
+    ffi::c_void,
+    io::{Read, Write},
+    os::raw::c_int,
+    pin::Pin,
+    task::Poll::Ready,
+};
+
+pub struct SignalToNoise {
```

Comment:
Hmmmmm still uncertain about this name. What about S2NHarness? Isn't that what this is? A collection of s2n objects that help you do performance testing? Sorry if I'm being too nitpicky about this, lmk what you think.

Reply:
The all caps does look a lot better, and I agree having `Harness` in the name makes sense because it's implementing a `TlsBenchHarness`. I don't know what the convention for capitalization of s2n is, but all caps `S2N` is better than `S2n` right?

Reply:
All-caps is fine to use. Yeah, struggling with the name. If we come up with a better name in the future we can always change it.